libraries/botbuilder-dialogs-adaptive/src/actions/switchCondition.ts

```diff
@@ -63,19 +62,55 @@ export class SwitchCondition<O extends object = {}> extends Dialog<O> implements
         }
 
         if (!this._caseExpresssions) {
-            this._caseExpresssions = {};
+            this._caseExpresssions = new Map<string, Expression>();
             for (let i = 0; i < this.cases.length; i++) {
                 const caseScope = this.cases[i];
-                const caseCondition = Expression.equalsExpression(this.condition, caseScope.createValueExpression());
-                this._caseExpresssions[caseScope.value] = caseCondition;
+                const intVal = parseInt(caseScope.value);
+                if (!isNaN(intVal)) {
+                    // you don't have to put quotes around numbers, "23" => 23 OR "23".
+                    this._caseExpresssions.set(caseScope.value, Expression.orExpression(
+                        Expression.equalsExpression(this.condition, new Constant(intVal)),
+                        Expression.equalsExpression(this.condition, new Constant(caseScope.value))
+                    ));
+                    continue;
+                }
+                const floatVal = parseFloat(caseScope.value);
+                if (!isNaN(floatVal)) {
+                    // you don't have to put quotes around numbers, "23" => 23 OR "23".
+                    this._caseExpresssions.set(caseScope.value, Expression.orExpression(
+                        Expression.equalsExpression(this.condition, new Constant(floatVal)),
+                        Expression.equalsExpression(this.condition, new Constant(caseScope.value))
+                    ));
+                    continue;
+                }
+                if (caseScope.value.trim().toLowerCase() === 'true') {
```

Comment:
May as well pull the result of `caseScope.value.trim().toLowerCase()` into a `const` above to avoid recomputation.